core/src/main/scala/org/apache/spark/scheduler/local/LocalSchedulerBackend.scala

```diff
@@ -110,7 +110,7 @@ private[spark] class LocalSchedulerBackend(
     val totalCores: Int)
   extends SchedulerBackend with ExecutorBackend with Logging {
 
-  private val appId = "local-" + System.currentTimeMillis
+  private val appId = sys.props.getOrElse("spark.test.appId", "local-" + System.currentTimeMillis)
```

Comment:
Is there a particular reason to retrieve this from the system properties instead of the SparkConf?

Reply:
Although I did to avoid `SparkConf(false)` case, in my use case, I always use as environment variables and it will be okay to load from `SparkConf`. Do you want me to change, @yaooqinn ?

Reply:
From my perspective, it looks better to read from SparkConf if it's available in this case.